### PR TITLE
fuy

### PR DESCRIPTION
**Pull request by fuywingfuy**
dddddd

### CODE REVIEW

File: BullsAndCows/BullsAndCowsGame.cs [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
-
+using System.Linq;
+
 namespace BullsAndCows
 {
     public class BullsAndCowsGame
     {
         private readonly SecretGenerator secretGenerator;
+        private readonly string secret = string.Empty;
         public BullsAndCowsGame(SecretGenerator secretGenerator)
```
Comment (fuywingfuy) [inline BullsAndCows/BullsAndCowsGame.cs:12]: vs

**Comment by YangJK2020** [comment]
good: baby step and readable message

File: BullsAndCowsTest/BullsAndCowsGameTest.cs [hunk 2]
```diff
@@ -8,10 +9,243 @@ public class BullsAndCowsGameTest
...
+            // var secretGenerator = new SecretGenerator();
+            var secretGenerator = new TestSecretGenerator();
             var game = new BullsAndCowsGame(secretGenerator);
             Assert.NotNull(game);
             Assert.True(game.CanContinue);
-        }
+        }
+
+        /*
+        [Fact]
```
Comment (YangJK2020) [inline BullsAndCowsTest/BullsAndCowsGameTest.cs:20]: Commented code should be deleted

File: BullsAndCowsTest/BullsAndCowsGameTest.cs [hunk 3]
```diff
@@ -8,10 +9,243 @@ public class BullsAndCowsGameTest
...
+        public void ShouldReturn0A0BGivenAllDigitAndWrongPosition()
+        {
+            // given
```
Comment (YangJK2020) [inline BullsAndCowsTest/BullsAndCowsGameTest.cs:23]: tests for secretGenerator and inputCheck should be implemented

File: BullsAndCows/BullsAndCowsGame.cs [hunk 4]
```diff
@@ -1,20 +1,73 @@
...
+        private readonly string errorInformation = "Wrong Input, input again";
         public BullsAndCowsGame(SecretGenerator secretGenerator)
         {
             this.secretGenerator = secretGenerator;
+            this.secret = this.secretGenerator.GenerateSecret();
         }
 
         public bool CanContinue => true;
 
+        public string GetMessage(string guess)
```
Comment (YangJK2020) [inline BullsAndCows/BullsAndCowsGame.cs:20]: should be CheckMessage

**Pull request closed without merging**